src/client/components/admin/UserTable.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import './UserTable.scss';
+
+interface UserTableProps {}
+
+class UserTable extends React.Component<{}, { userData: Array<React.ReactElement<any>> }> {
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            userData: [],
+        };
+    }
+
+    public async componentDidMount(): Promise<void> {
+        const result = await fetch('/graphql', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+                query: '{users { user_id, name, email, user_class }}',
+            }),
+        });
+        const data = await result.json();
+        const users = data.data.users;
+        const newUserData: Array<React.ReactElement<any>> = [];
+        for (let i = 0; i < users.length; i++) {
+            newUserData.push(<UserTableRow key={i} user_id={users[i].user_id} name={users[i].name} email={users[i].email} user_class={users[i].user_class} />);
+        }
+
+        this.setState({userData: newUserData});
```

Comment:
You're calling setState in ComponentDidMount which happens after render in the component lifecycle?

Reply:
Not sure if best practice, but the component renders nothing (or loader) until the data is received, which triggers an update. 
> Calling setState() in this method will trigger an extra rendering, but it will happen before the browser updates the screen. This guarantees that even though the render() will be called twice in this case, the user won’t see the intermediate state.